src/sources/aws_ecs_metrics/mod.rs

```diff
@@ -154,11 +155,10 @@ async fn aws_ecs_metrics(
                         match parser::parse(body.as_ref(), namespace.clone()) {
                             Ok(metrics) => {
                                 let count = metrics.len();
-                                emit!(&HttpEventsReceived {
```

Comment:
I'd also be content to have `HttpEventsReceived` emit the deprecated metrics (even if that means it adds them in places they weren't published before) so you don't need to have the custom event here.

Reply:
This case of dropping an existing event was a mistake. I take care not to drop them usually. Therefore the other sources should not be impacted.
I see the nice part of moving the deprecated metrics to the global event, but this would add so useless calls to every sources and might slow down Vector.

Reply:
Makes sense, thanks!